src/main/webapp/js/simulations.js

```diff
@@ -55,26 +55,53 @@ function addSimulationRows(table, simulations) {
     for (const id in simulations) {
         const simulation = simulations[id];
         var row = table.insertRow(-1);
-        
-        var visualizeSimulationButton = document.createElement('button');
-        visualizeSimulationButton.innerText = 'Visualize Simulation';
-        /**
-         * For each simulation add a button redirecting to simulation_visualization.html
-         * Simulation metadata is encoded as a query string.
-         */
-        visualizeSimulationButton.addEventListener('click', () => {
-            // Create a deep copy of simulation JSON, and the simulation id as a new key.
-            var simulationWithId = JSON.parse(JSON.stringify(simulation));
-            simulationWithId['id'] = id;
-            
-            // Redirect.
-            window.location.replace('simulation_visualization.html?' + toQueryString(simulationWithId));
-        });
-
-        row.insertCell(0).appendChild(visualizeSimulationButton);
+        const visualizeSimulationButton = createVisualizationButton(simulation, id);
+        const deleteSimulationButton = createDeletionButton(id);
+        const cell = row.insertCell(0);
+        cell.appendChild(visualizeSimulationButton);
+        cell.appendChild(deleteSimulationButton);
```

Comment:
Aren't the buttons too close? We don't want the user to accidentally delete the simulation they want to see.

Reply:
Clicking the delete button opens a confirmation box, so I think that's fine (this can be modified if visual design is the issue, of course).